src/CSV.jl

```diff
@@ -61,6 +61,41 @@ const EMPTY_TYPEMAP = Dict{TypeCode, TypeCode}()
 const EMPTY_REFS = Vector{String}[]
 const EMPTY_REFVALUES = String[]
 
+const INVALID_DELIMITERS = ['\r', '\n', '\0']
+
+"""
+    isvaliddelim(delim)
+
+Whether a character or string is valid for use as a delimiter.
+"""
+isvaliddelim(delim) = false
+isvaliddelim(delim::Char) = delim ∉ INVALID_DELIMITERS
+isvaliddelim(delim::AbstractString) = all(isvaliddelim, delim)
+
+"""
+    checkvaliddelim(delim)
+
+Checks whether a character or string is valid for use as a delimiter.  If
+`delim` is `nothing`, it is assumed that the delimiter will be auto-selected.
+Throws an error if `delim` is invalid.
+"""
+function checkvaliddelim(delim)
+    delim ≢ nothing && !isvaliddelim(delim) &&
+        throw(ArgumentError("invalid delim argument = '$(escape_string(string(delim)))', "*
+                            "the following delimiters are invalid: $INVALID_DELIMITERS"))
+end
+
+"""
+    checkvalidsource(source)
+
+Checks whether the argument is valid for use as a data source, otherwise throws
+an error.
+"""
+function checkvalidsource(source)
+    !isa(source, IO) && !isa(source, Vector{UInt8}) && !isfile(source) &&
```

Comment:
Can you update this to account for https://github.com/JuliaData/CSV.jl/pull/473 that I just merged?

Reply:
Done